src/pipx/venv.py

```diff
@@ -173,32 +176,25 @@ def install_package(
         is_main_package: bool,
     ) -> None:
 
-        with animate(f"installing package {package_or_url!r}", self.do_animation):
-            if pip_args is None:
-                pip_args = []
+        if pip_args is None:
+            pip_args = []
+        if package is None:
+            # If no package name is supplied, install only main package
+            #   first in order to see what its name is
+            package = self.install_package_no_deps(package_or_url, pip_args)
             if package is None:
-                # If no package name is supplied, install only main package
-                #   first in order to see what its name is
-                old_package_set = self.list_installed_packages()
-                cmd = ["install"] + pip_args + ["--no-dependencies"] + [package_or_url]
-                self._run_pip(cmd)
-                installed_packages = self.list_installed_packages() - old_package_set
-                if len(installed_packages) == 1:
-                    package = installed_packages.pop()
-                    logging.info(f"Determined package name: '{package}'")
-                else:
-                    package = None
+                logging.warning(
+                    f"Cannot determine package name for spec {package_or_url!r}. "
+                )
+                raise PackageInstallFailureError
```

Comment:
I agree an error should be raised if the package name cannot be determined. However, when a `PackageInstallFailureError` is raised, we could propagate more specific failure information to the user instead of unconditionally printing this message:
```
        except PackageInstallFailureError:
            venv.remove_venv()
            raise PipxError(
                f"Could not install package {package_name}. Is the name or spec correct?"
            )
```

I think it would be better if we could raise `PackageInstallFailureError` with specific failure text, and print that text directly to the user. i.e. `raise PackageInstallFailureError( f"Cannot determine package name for spec {package_or_url!r}.")` and then in `main.py` do

```
    except (PipxError, PackageInstallFailureError) as e:
```